### PR TITLE
chore(#9081): adds environment shared lib

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9081-environment-shared-library/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9081-environment-shared-library/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9081-environment-shared-library/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Hi @jkuester . Would you mind a review for this shared library? 

I am going to implement using sessions for all request-promise-native interactions with the database, but that meant adding yet another place where we parse and process environment variables. This is an intermediary step towards that goal.

**Comment by jkuester** [comment]
Saw the 82 files changed and got scared, but then I see `+381 −273` and am not so afraid... :sweat_smile: 

Plan to have a look at this today or tomorrow at the latest.

File: shared-libs/environment/test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+const chai = require('chai');
+const expect = chai.expect;
```
Comment (jkuester) [inline shared-libs/environment/test/index.spec.js:2]:
```suggestion
const { expect } = require('chai');
```

File: shared-libs/environment/test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+describe('environment', () => {
+  afterEach(() => {
+    sinon.restore();
+    process.env = originalEnv;
+  });
+
+  it('should exit if no couchurl', async () => {
+    sinon.stub(process, 'exit');
+    stubProcessEnv({ COUCH_URL: '' });
+    rewire('../src/index');
```
Comment (jkuester) [inline shared-libs/environment/test/index.spec.js:21]: Interesting use of `rewire`.  Nice!

File: shared-libs/environment/test/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    expect(process.exit.calledWith(1)).to.be.true;
```
Comment (jkuester) [inline shared-libs/environment/test/index.spec.js:22]: Minor, but we should also be able to also stub and assert the logger in these first two tests, right?

File: api/tests/mocha/resources.spec.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const rewire = require('rewire');
+const { expect } = require('chai');
+const path = require('path');
+
+let resources;
+
+describe('resources', () => {
+  beforeEach(() => {
+    resources = rewire('../../src/resources.js');
```
Comment (jkuester) [inline api/tests/mocha/resources.spec.js:9]: Minor, but if we are not going to change anything between tests, is there any reason that we actually need `rewire`?

Comment (jkuester) [reply api/tests/mocha/resources.spec.js:9]: Ahh, I see now. This file is just renamed from `environment.spec.js`, so you did not write it.  Well, no big deal either way if we change this or not...

File: sentinel/src/db.js [hunk 5]
```diff
@@ -51,18 +52,15 @@ if (UNIT_TEST_ENV) {
...
   module.exports.queryMedic = stubMe('queryMedic');
-} else if (COUCH_URL) {
+} else {
   const PouchDB = require('pouchdb-core');
   PouchDB.plugin(require('pouchdb-adapter-http'));
   PouchDB.plugin(require('pouchdb-session-authentication'));
   PouchDB.plugin(require('pouchdb-mapreduce'));
   PouchDB.plugin(require('pouchdb-replication'));
 
   // strip trailing slash from to prevent bugs in path matching
```
Comment (jkuester) [inline sentinel/src/db.js:62]:
```suggestion
```

**Review by jkuester — APPROVED** [review]
So nice to get this centralized!  LGTM!

**Pull request merged**